src/transform.jl

```diff
@@ -206,38 +206,30 @@ const PDMatDistribution = Union{InverseWishart, Wishart}
 
 function link(d::PDMatDistribution, X::AbstractMatrix{T}) where {T<:Real}
   Y = cholesky(X).L
-  dim = size(Y)
-  for m in 1:dim[1]
+  for m in 1:size(Y, 1)
     Y[m, m] = log(Y[m, m])
   end
-  for m in 1:dim[1], n in m+1:dim[2]
-    Y[m, n] = zero(T)
-  end
-  Matrix{T}(Y)
+  return Y
 end
 
-###### FIX THIS. IT'S BROKEN!
-function invlink(d::PDMatDistribution, Z::AbstractMatrix{<:Real}) where {T<:Real}
-  dim = size(z)
-  for m in 1:dim[1]
-    z[m, m] = exp.(z[m, m])
-  end
-  for m in 1:dim[1], n in m+1:dim[2]
-    z[m, n] = zero(T)
+function invlink(d::PDMatDistribution, Y::LowerTriangular{T}) where {T<:Real}
+  X, dim = copy(Y), size(Y)
```

Comment:
Do you intend to write `zero(Y)`

Reply:
I don't believe so. Previously we were just mutating `Y` directly, so I think that `copy`ing is the appropriate behaviour.

Reply:
It seems that we don't have the L225-L227 of the original code.

Reply:
Thanks for the clarification. I think here should be no problem.

Reply:
Okay. I'll dig into the code a little deeper and try to figure out what's going wrong.

Reply:
Okay, I think that I've figured out what the root cause of the problem is here. The story is as follows:
1. With `PDMatDistribution`s, I want their unconstrained form to be `LowerTriangular` type, and the constrained type to be `Matrix` (really I want it to be `PDMat`, but that's another discussion). 
2. [link](https://github.com/TuringLang/Turing.jl/blob/e6782521706f8343ea7e364cb87537461b6cc13b/src/core/varinfo.jl#L125) and [invlink](https://github.com/TuringLang/Turing.jl/blob/e6782521706f8343ea7e364cb87537461b6cc13b/src/core/varinfo.jl#L139) require that the manner in which `reconstruct` and `vectorize` function depend only upon the distribution `dist` and a vector of `Real`s pulled from a `var_info` object. Crucially, this means that the particular method of `reconstruct` used _cannot_ depend on whether the thing being reconstructed was previously a `LowerTriangular` or a `Matrix`. This also explains why the previous implementation worked nicely: the matrix ways always represented as a `Matrix`, never a `LowerTriangular`, meaning that this simply wasn't an issue.

I'll find a hack for now, but we should really address this - moving forward it will become problematic if we have to keep hacking our way around this.